src/Nethermind/Nethermind.Trie/Pruning/ITrieStoreExtensions.cs

```diff
@@ -1,12 +1,17 @@
 // SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
+using Nethermind.Core.Crypto;
+
 namespace Nethermind.Trie.Pruning
 {
     // ReSharper disable once InconsistentNaming
     public static class ITrieStoreExtensions
     {
         public static IReadOnlyTrieStore AsReadOnly(this ITrieStore trieStore, INodeStorage? readOnlyStore = null) =>
             trieStore.AsReadOnly(readOnlyStore);
+
+        public static IScopedTrieStore GetTrieStore(this ITrieStore trieStore, ValueHash256 address) =>
```

Comment:
Should be removed so that less mistake on top level where the address is stored as `ValueHash256` which may result in increase memory usage in pruning cache.

Reply:
So besides tests it has 3 usages and all of them allocated before `Hash256` from `Address.ToAccountPath`. I think we can't easily avoid allocating here. I can remove but the result will always be the same + like 20 explicit allocations (or 3 if I make this method available in tests).

Reply:
Changed it to take `Address`